nsqd/channel.go

```diff
@@ -389,15 +389,22 @@ func (c *Channel) RequeueMessage(clientID int64, id MessageID, timeout time.Dura
 }
 
 // AddClient adds a client to the Channel's client list
-func (c *Channel) AddClient(clientID int64, client Consumer) {
+func (c *Channel) AddClient(clientID int64, client Consumer) error {
 	c.Lock()
 	defer c.Unlock()
 
 	_, ok := c.clients[clientID]
 	if ok {
-		return
+		return nil
+	}
+
+	maxChannelConsumers := c.ctx.nsqd.getOpts().MaxChannelConsumers
+	if maxChannelConsumers == 0 || (maxChannelConsumers != 0 && len(c.clients) < maxChannelConsumers) {
+		c.clients[clientID] = client
+		return nil
 	}
-	c.clients[clientID] = client
+
+	return errors.New("E_TOO_MANY_CHANNEL_SUBSCRIPTIONS")
```

Comment:
style nit-pick, I'd prefer the non-error case to be the default path and the error case to be the branch, please :)

Reply:
Will do.